test/aggregates/aggregates_integration_test.go

```diff
@@ -0,0 +1,260 @@
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+
+package aggregates
+
+import (
+	"fmt"
+	"github.com/v3io/v3io-tsdb/pkg/config"
+	"github.com/v3io/v3io-tsdb/pkg/tsdb"
+	"github.com/v3io/v3io-tsdb/pkg/tsdb/tsdbtest"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+	"testing"
+	"time"
+)
+
+type TestConfig struct {
+	desc          string
+	testStartTime int64
+	testBaseTime  int64
+	interval      int64
+	testDuration  int64
+	numMetrics    int
+	numLabels     int
+	values        []float64
+	queryFunc     string
+	queryStart    int64
+	queryStep     int64
+	expectedCount int
+	expectedSum   float64
+	expectedAvg   float64
+	expectFail    bool
+	ignoreReason  string
+	v3ioConfig    *config.V3ioConfig
+	setup         func() (*tsdb.V3ioAdapter, func())
+}
+
+type metricContext struct {
+	lset utils.Labels
+	ref  uint64
+}
+
+func TestAggregates(t *testing.T) {
+
+	testCases := []*TestConfig{
+		t1Config(t),
+	}
+
+	for _, testConfig := range testCases {
+		t.Logf("%s\n", testConfig.desc)
+		t.Run(testConfig.desc, func(t *testing.T) {
+			if testConfig.ignoreReason != "" {
+				t.Skip(testConfig.ignoreReason)
+			}
+			testAggregatesCase(t, testConfig)
+		})
+	}
+
+}
+
+func t1Config(testCtx *testing.T) *TestConfig {
+	currentTimeNano := time.Now().UnixNano()
+
+	testDuration := int64(80 * time.Hour)
+
+	s := &TestConfig{
+		desc:          "Test case #1",
+		testStartTime: currentTimeNano,
+		testBaseTime:  currentTimeNano - testDuration - int64(time.Minute),
+		interval:      int64(10 * time.Second),
+		testDuration:  testDuration,
+		numMetrics:    1,
+		numLabels:     1,
+		values:        []float64{1, 2, 3, 4, 5},
+		queryFunc:     "count,sum,avg",
+		queryStart:    currentTimeNano - int64(88*time.Hour),
+		queryStep:     int64(1 * time.Hour),
+		setup: func() (*tsdb.V3ioAdapter, func()) {
+			return setupFunc(testCtx)
+		},
+	}
+	return s
+}
+
+// Setup for test and return tear-down function
+func setupFunc(testCtx *testing.T) (*tsdb.V3ioAdapter, func()) {
+	// setup:
+	v3ioConfig, err := tsdbtest.LoadV3ioConfig()
+	if err != nil {
+		testCtx.Fatalf("Failed to read config %s", err)
+	}
+	v3ioConfig.Path = fmt.Sprintf("%s-%d", testCtx.Name(), time.Now().Nanosecond())
+
+	tsdbtest.CreateTestTSDB(testCtx, v3ioConfig)
+
+	adapter, err := tsdb.NewV3ioAdapter(v3ioConfig, nil, nil)
+	if err != nil {
+		testCtx.Fatal(err)
+	}
+
+	return adapter, func() {
+		// Tear down:
+		// Don't delete the table if the test has failed
+		if !testCtx.Failed() {
+			tsdbtest.DeleteTSDB(testCtx, v3ioConfig)
+		}
+	}
+}
+
+func testAggregatesCase(t *testing.T, testConfig *TestConfig) {
+	// Setup & TearDown
+	adapter, tearDown := testConfig.setup()
+	defer tearDown()
+
+	startBefore := testConfig.testStartTime - testConfig.queryStart - int64(1*time.Minute)
```

Comment:
makes no sense, it's basically starting query from `int64(88*time.Hour) -  int64(1*time.Minute)`